xmtp_id/src/lib.rs

```diff
@@ -43,19 +51,50 @@ impl InboxOwner for LocalWallet {
     }
 }
 
+pub struct Credential {
+    pub(crate) inbox_id: String,
+}
+
+impl From<Credential> for MlsCredentialProto {
+    fn from(credential: Credential) -> Self {
+        MlsCredentialProto {
+            inbox_id: credential.inbox_id,
+        }
+    }
+}
+
+impl From<MlsCredentialProto> for Credential {
+    fn from(credential: MlsCredentialProto) -> Self {
+        Credential {
+            inbox_id: credential.inbox_id,
+        }
+    }
+}
+
 /// XMTP Identity according to [XIP-46](https://github.com/xmtp/XIPs/pull/53)
 pub struct Identity {
     #[allow(dead_code)]
-    id: String,
+    id: String, // inbox id
+    #[allow(dead_code)]
+    credential: RwLock<Option<OpenMlsCredential>>,
 }
 
 impl Identity {
     /// Generate a new, empty ID for an account address.
     /// A nonce is used to ensure uniqueness of the ID.
-    pub fn new(address: String) -> Self {
+    pub fn new(
+        address: String,
+        nonce: &u64,
+        credential: Credential,
```

Comment:
It seems like we have everything needed to generate the credential from the address and nonce

Reply:
Makes sense. I added a `create_credential()` func, I think that's needed to interface with open mls?